src/core/arm/arm_interface.h

```diff
@@ -189,4 +238,67 @@ class ARM_Interface : NonCopyable {
 
 private:
     u32 id;
+
+    friend class boost::serialization::access;
+
+    template <class Archive>
+    void save(Archive& ar, const unsigned int file_version) const {
+        ar << timer;
+        ar << id;
+        auto page_table = GetPageTable();
+        ar << page_table;
+        for (auto i = 0; i < 15; i++) {
+            auto r = GetReg(i);
+            ar << r;
+        }
+        auto pc = GetPC();
```

Comment:
```suggestion
        const auto pc = GetPC();
```

etc, where applicable below